### ISSUE TITLE
Question regarding interpretation of output

### ISSUE THREAD

**Issue by MUppal** [AlexandrovLab/SigProfilerExtractor#68]
Hi,

I have a question regarding appropriate interpretation of results for your tool, in particular identifying the extracted signatures as reference COSMIC signatures vs. de novo signatures. 

For example, in my dataset consisting of lung SNVs across a few hundred samples, it appears there are 4 active signatures based off the SBS96_selection_plot. When looking at the file De_Novo_map_to_COSMIC_SBS96.csv, I see a percentage breakdown of COSMIC signatures as enriched in these 4 extracted signatures (see below). Here is where I am unclear on how use this information to identify the extracted signatures as identical to the reference COSMIC signatures vs de novo.

![Picture1](https://user-images.githubusercontent.com/10904116/109901312-d65f2480-7c66-11eb-9c07-57cf13174c61.png)

My understanding is 1) when looking at the second column, if there is no COMSIC signature enrichment, I can interpret that as a novel signature extracted, 2) in the event that there’s a high percentage of enrichment (eg. >80%), then that extracted signature is uniquely that reference COSMIC signature.

But I am unsure on how to proceed in the case where enrichment of COSMIC signatures is intermediate. For example, in Signature 96-D in the table above, a number of enriched signatures do not surpass 50%, but still appear to have appreciable overlap with the extracted signature (say, 20-40%). Am I to assume that these enriched COSMIC signatures are actually active and contributing to what appears to be the single extracted signature (Sig 96-D)? Or does that lack of dominance of a single reference signature indicate that Sig 96-D is a unique and novel signature? In the event of the former, going through all the COSMIC signatures enriched >10% in the whole table would amount to >4 signatures as originally identified.

I look forward to hearing your insight on how to appropriately interpret this output. Thank you.

**Comment by mishugeb** [comment]
Hi MUppal,
The purpose to decompose the de novo signature to the COSMIC signatures is to see if the de novo signatures are already identified previously.  A de novo signature is decomposed to the COSMIC signatures using an algorithm described in https://www.biorxiv.org/content/10.1101/2020.12.13.422570v1.abstract. If less than five COSMIC signatures are mapped to a de novo signature with a cosine similarity of more than 0.85, then that is not a new signature. Otherwise, the de novo signature is considered a novel signature. The individual percentage of the decomposed  COSMIC signature doesn't matter. 

Thanks,
Mishu

**Comment by MUppal** [comment]
Hi Mishu,

Thank you for the insight and the link to the paper, it was very helpful. My apologies for the delay in my response and in re-opening the issue. I am still in need of some further clarification. Based off what you said, if a given extracted signature, say Sig 96-D in my post above, has a cosine similarity > 0.85 (it has 0.9), then I can interpret that as a previously characterized signature. However, I'm having difficulty defining which exact signature. Am I to interpret that the 4 enriched signatures SBS5, SBS24, SBS30, and SBS36 are all active in my cohort? Or is it only a single one of those signatures that can be uniquely identified with Sig 96-A? If it's the former, then applying that rule to the other extracted signatures would imply that there are > 4 active signatures in my cohort. For example, doing so would imply 11 active COSMIC signatures throughout the whole cohort as represented in COSMIC_SBS96_Activities_refit.txt. Can you advise as to how I can take the next step to find the exact activities of both previously characterized COSMIC signatures and true de novo signatures in my set of variants?

**Comment by mishugeb** [comment]
Hi Manik,
In the results folder, you will get a subfolder named "suggested solutions". In that one, you get two folders with COSMIC results and De Novo results. De Novo result will contain the de novo signatures and their activities on the samples. Further, we try to map the COSMIC signatures to each of the De Novo signatures to see if the De Novo signatures can be explained by one or more(up to 4) COSMIC signatures in a linear combination within a certain cosine similarity threshold (0.9). So if you find a De Novo signature is decomposed into 4 COSMIC signatures, then you will assume the 4 signatures are present in the cohort. However, during the final assignment (you will find that COSMIC signature results directory) some COSMIC signatures may have 0 contributions to the samples. So, please follow the final activities of the COSMIC signatures in samples in the activity file to see what COSMIC signatures are present on the samples.

Thanks,
Mishu

**Comment by MUppal** [comment]
Hi Mishu,

Following up regarding your last couple points. How exactly then should one combine the results from decomposed and de novo solutions?

For example, let’s say in the file De_Novo_map_to_COSMIC_SBS96.csv, there is Signature 96-D which has the following breakdown:  Signature SBS5 (10.24%) & Signature SBS29 (89.76%), which has a cosine similarity of 0.84 (which we established above is below the cutoff, and must be considered a de novo signature). In SBS96_De-Novo_Activities_refit.txt, the third sample in the cohort features 27 variants assigned to SBS96D. However, in COSMIC_SBS96_Activities_refit.txt there are 0 variants assigned to SBS29.  

Clearly it would be incorrect to combine refit and de novo activities by subtracting off the number of counts assigned to de novo signatures from the refit (weighted by their respective contributions to the signature) - in this case that would results in -24 counts of SBS29 for that sample. 

So how exactly do we combine the 2 to get a final list of the number of known COSMIC signatures and de novo signatures for each sample?

**Comment by mishugeb** [comment]
I am a little confused about the question. @lalexandrov1018 could you please see if that makes sense to you? If does, could you please give us a comment on the problem?

**Comment by MUppal** [comment]
Hi Mishu,

My apologies for the lack of clarity in my prior post. I hope I’ve done a better job in this one. To state very clearly at the outset, my goal is to construct a matrix like COSMIC_SBS96_Activities_refit.txt with counts of the active COSMIC signatures and appropriately identified de novo signatures. The problem that I’m facing is outlined as follows:

1. Using the output from the first post as an example, from my De_Novo_map_to_COSMIC_SBS96.csv file, the decomposition of Signature 96-B has a cosine similarity of 0.84, which is <0.85, which we established earlier is our cutoff for defining this as actually a de novo signature.
2. If I go to COSMIC_SBS96_Activities_refit.txt, there are counts present for all 11 COSMIC SBS signatures identified in the column “Global NMF Signatures” in De_Novo_map_to_COSMIC_SBS96.csv. In particular, many samples have counts attributed to SBS17a and SBS26, which are present from the decomposition of Signature 96-B.
3. However, based on point 1), I must treat Signature 96-B as its own de novo signature, and so SBS17a and SBS26 won’t be present in my final activities matrix (which will contain the active SBS signatures and the de novo signatures). 
4. The issue I’m running into generally derives from your line in your second post above, “during the final assignment … some COSMIC signatures may have 0 contributions to the samples.” This is illustrated in the different example output I gave in my third post - in that example, a “de novo” signature has 27 counts assigned for a particular sample in SBS96_De-Novo_Activities_refit.txt, however the COSMIC signature that has the largest contribution (SBS29 in that example) has 0 counts for that sample in the COSMIC_SBS96_Activities_refit.txt file. If my goal is to reassign 27 of the COSMIC counts in COSMIC_SBS96_Activities_refit.txt that are derived from the de novo signature, then I can’t do so by subtracting off the weighted contribution of the COSMIC signatures to de novo signature (for SBS29, the weighted contribution of counts would be 0.89 * 27 = 24, however the COSMIC_SBS96_Activities_refit.txt features 0 counts for SBS29). Clearly, the tool’s process of refitting activities makes translating between SBS96_De-Novo_Activities_refit.txt and COSMIC_SBS96_Activities_refit.txt difficult.
5. I’ll give another example of the difficulty I’m having in translating between the two activities files by using the example output described in bullets 1-3 and the very first post in this issue. For a given sample, SBS96_De-Novo_Activities_refit.txt features 488 counts attributed to Signature 96-B, which we’ve established must be treated as a de novo signature. In COSMIC_SBS96_Activities_refit.txt, there are 419 counts attributed to SBS26 and 157 attributed to SBS17a. This is in total 576 counts attributable to COSMIC signatures that comprise a de novo signature that has only 488 assigned counts for that sample (no other signatures are comprised of SBS17a and SBS26). So again, the process of refitting each sample with the COSMIC signatures results in discrepancies between quantifications in SBS96_De-Novo_Activities_refit.txt and COSMIC_SBS96_Activities_refit.txt. 
6. Again, my goal is to construct a matrix like COSMIC_SBS96_Activities_refit.txt with the active COSMIC signatures and appropriately identified de novo signatures. Given the impossibility of translating between the De-Novo activities file and COSMIC activities file, how should I go about obtaining my desired matrix?

And of course, the difficulties I’m having beg the question of if I’m even understanding and using the output of the tool correctly. How exactly should I be making use of the De-Novo and COSMIC activities files to achieve my goal? Should I actually only be treating as de novo signatures the ones described here: https://github.com/AlexandrovLab/SigProfilerExtractor/issues/26 , where extracted signatures could not be mapped to COSMIC signature at all? Should I not be concerned at all with extracted signatures with cosine similarities < 0.85 in De_Novo_map_to_COSMIC_SBS96.csv ?
